public/app/plugins/panel/alertlist/unified-alerting/UngroupedView.tsx

```diff
@@ -27,86 +31,176 @@ type UngroupedModeProps = {
   options: UnifiedAlertListOptions;
 };
 
+export const useGetExtraInfoForAlertItem = (returnTo: string, slicedRules: CombinedRuleWithLocation[]) => {
+  const [rulesToDisplay, setRulesToDisplay] = useState<CombinedRuleWithLocationExtended[] | undefined>(undefined);
+  useEffect(() => {
+    const getExtraInfo = async () => {
+      const rulesWithExtraInfo: CombinedRuleWithLocationExtended[] = await Promise.all(
+        slicedRules.map(async (rule: CombinedRuleWithLocation) => {
+          const identifier = fromCombinedRule(rule.dataSourceName, rule);
+          const strIndentifier = stringifyIdentifier(identifier);
+
+          const refToRuleView = createUrl(
+            `/alerting/${encodeURIComponent(rule.dataSourceName)}/${encodeURIComponent(strIndentifier)}/view`,
+            { returnTo: returnTo ?? '' }
+          );
+          type HistoryResponse = {
```

Comment:
what about defining this `type` outside of this function? Just in case it needs to be used somewhere else at some point.